src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractDatasetCommand.java

```diff
@@ -155,32 +155,30 @@ protected void tidyUpFields(DatasetVersion dsv) {
      * @param ctxt
      * @throws CommandException
      */
-    protected void registerExternalIdentifier(Dataset theDataset, CommandContext ctxt) throws CommandException {
+    protected void registerExternalIdentifier(Dataset theDataset, CommandContext ctxt, boolean retry) throws CommandException {
         if (!theDataset.isIdentifierRegistered()) {
             GlobalIdServiceBean globalIdServiceBean = GlobalIdServiceBean.getBean(theDataset.getProtocol(), ctxt);
             if ( globalIdServiceBean != null ) {
                 if (globalIdServiceBean instanceof FakePidProviderServiceBean) {
-                    try {
-                        globalIdServiceBean.createIdentifier(theDataset);
-                    } catch (Throwable ex) {
-                        logger.warning("Problem running createIdentifier for FakePidProvider: " + ex);
-                    }
-                    theDataset.setGlobalIdCreateTime(getTimestamp());
-                    theDataset.setIdentifierRegistered(true);
-                    return;
+                    retry=false; //No reason to allow a retry with the FakeProvider, so set false for efficiency
                 }
                 try {
                     if (globalIdServiceBean.alreadyExists(theDataset)) {
                         int attempts = 0;
-
-                        while (globalIdServiceBean.alreadyExists(theDataset) && attempts < FOOLPROOF_RETRIAL_ATTEMPTS_LIMIT) {
-                            theDataset.setIdentifier(ctxt.datasets().generateDatasetIdentifier(theDataset, globalIdServiceBean));
-                            logger.log(Level.INFO, "Attempting to register external identifier for dataset {0} (trying: {1}).",
-                                new Object[]{theDataset.getId(), theDataset.getIdentifier()});
-                            attempts++;
+                        if(retry) {
+                            do  {
+                                theDataset.setIdentifier(ctxt.datasets().generateDatasetIdentifier(theDataset, globalIdServiceBean));
+                                logger.log(Level.INFO, "Attempting to register external identifier for dataset {0} (trying: {1}).",
+                                    new Object[]{theDataset.getId(), theDataset.getIdentifier()});
+                                attempts++;
+                            } while (globalIdServiceBean.alreadyExists(theDataset) && attempts <= FOOLPROOF_RETRIAL_ATTEMPTS_LIMIT);
                         }
-
-                        if (globalIdServiceBean.alreadyExists(theDataset)) {
```

Comment:
Similar - extra call to alreadyExists since attempts will be over the limit if the while call above doesn't find alreadyExists()==false by the end.